xayn-ai/src/coi/system.rs

```diff
@@ -81,35 +81,34 @@ impl systems::CoiSystem for CoiSystem {
 /// Assigns a CoI for the given embedding.
 ///
 /// Returns `None` if no CoI could be found otherwise it returns the Id of
-/// the CoL along with the positive and negative distance. The negative distance
-/// will be [`f32::MAX`], if no negative coi could be found.
+/// the CoL along with the positive and negative similarity.
 pub(crate) fn compute_coi_for_embedding(
     embedding: &Embedding,
     user_interests: &UserInterests,
-    neighbors: usize,
 ) -> Option<CoiComponent> {
-    let (coi, pos_distance) = find_closest_coi(embedding, &user_interests.positive, neighbors)?;
-    let neg_distance = match find_closest_coi(embedding, &user_interests.negative, neighbors) {
+    let (coi, pos_similarity) = find_closest_coi(embedding, &user_interests.positive)?;
+    let neg_similarity = match find_closest_coi(embedding, &user_interests.negative) {
         Some((_, dis)) => dis,
-        None => f32::MAX,
+        // TODO: Is this correct? This was previously f32::MAX, but given that we
+        //       now have a similarity score instead of a distance, I'd assume it should be 0.
```

Comment:
i'd say it should be `-1` as this is the lowest possible similarity, equivalently to the previous highest distance.